src/mme-app/actionHandlers/serviceRequestActionHandlers.cpp

```diff
@@ -303,6 +318,46 @@ ActStatus ActionHandlers::process_mb_resp_svc_req(ControlBlock& cb)
 	return ActStatus::PROCEED;
 }
 
+/***************************************
+* Action handler : send_service_reject
+***************************************/
+ActStatus ActionHandlers::send_service_reject(ControlBlock& cb)
+{
+  	log_msg(LOG_DEBUG, "Inside send_service_reject \n");
+        
+	UEContext *ue_ctxt = static_cast<UEContext*>(cb.getPermDataBlock());
+        if (ue_ctxt == NULL)
+        {
+                log_msg(LOG_DEBUG, "send_service_reject: ue context is NULL \n");
+                return ActStatus::HALT;
+        }
+
+	MmeProcedureCtxt *procCtxt = dynamic_cast<MmeProcedureCtxt*>(cb.getTempDataBlock());
+	if (procCtxt == NULL)
+        {
+                log_msg(LOG_DEBUG, "send_service_reject: procedure ctxt is NULL \n");
+                return ActStatus::HALT;
+        }
+
+        
+	struct commonRej_info service_rej;
+	
+	service_rej.msg_type = service_reject;
+	service_rej.ue_idx = ue_ctxt->getContextID(); 
+	service_rej.s1ap_enb_ue_id = ue_ctxt->getS1apEnbUeId();
+	service_rej.enb_fd = ue_ctxt->getEnbFd();
+	service_rej.cause = emmCause_ue_id_not_derived_by_network;
+	
+	cmn::ipc::IpcAddress destAddr;
```

Comment:
use constructor to initialize value.

Reply:
Will do this as part of message data struct cleanup.